test/NuGet.Core.FuncTests/NuGet.Core.FuncTest/DownloadTimeoutStreamTests.cs

```diff
@@ -64,76 +64,82 @@ public async Task DownloadTimeoutStream_ApmNotSupported()
-            
+
             // Act & Assert
             await Assert.ThrowsAsync<NotSupportedException>(() =>
                 ReadStreamApm(timeoutStream));
         }
 #endif
-        
+
         [Fact]
         public async Task DownloadTimeoutStream_SuccessSync()
         {
             await VerifySuccessfulReadAsync(stream => Task.FromResult(ReadStream(stream)));
         }
-        
+
         [Fact]
         public async Task DownloadTimeoutStream_TimeoutSync()
         {
             await VerifyTimeoutOnReadAsync(stream => Task.FromResult(ReadStream(stream)));
         }
-        
+
         [Fact]
         public async Task DownloadTimeoutStream_FailureSync()
         {
             await VerifyFailureOnReadAsync(stream => Task.FromResult(ReadStream(stream)));
         }
-        
+
         public async Task VerifyFailureOnReadAsync(Func<Stream, Task<string>> readAsync)
         {
             // Arrange
             var expected = new IOException();
             var memoryStream = GetStream("foobar");
             var slowStream = new SlowStream(memoryStream)
             {
-                OnRead = (buffer, offset, count) => { throw expected; } 
+                OnRead = (buffer, offset, count) => { throw expected; }
             };
             var timeoutStream = new DownloadTimeoutStream(
                 "download",
                 slowStream,
                 TimeSpan.FromSeconds(10));
-            
+
             // Act & Assert
             var actual = await Assert.ThrowsAsync<IOException>(() =>
                 readAsync(timeoutStream));
             Assert.Same(expected, actual);
         }
-        
+
         public async Task VerifyTimeoutOnReadAsync(Func<Stream, Task<string>> readAsync)
         {
             // Arrange
             var expectedDownload = "download";
-            var expectedMilliseconds = 25;
+            var timeout = TimeSpan.FromMilliseconds(25);
             var memoryStream = GetStream("foobar");
-            var slowStream = new SlowStream(memoryStream)
+
+            using (var cancellationTokenSource = new CancellationTokenSource())
             {
-                DelayPerByte = TimeSpan.FromMilliseconds(expectedMilliseconds * 2)
-            };
-            var timeoutStream = new DownloadTimeoutStream(
-                expectedDownload,
-                slowStream,
-                TimeSpan.FromMilliseconds(expectedMilliseconds));
-            
-            // Act & Assert
-            var exception = await Assert.ThrowsAsync<IOException>(() =>
-                readAsync(timeoutStream));
-            Assert.Equal(
-                $"The download of '{expectedDownload}' timed out because " +
-                $"no data was received for {expectedMilliseconds}ms.",
-                exception.Message);
-            Assert.IsType<TimeoutException>(exception.InnerException);
+                var slowStream = new SlowStream(memoryStream)
```

Comment:
Did you intend to use the new `SlowStream` constructor to pass the cancellation token? If not, the cancellation token source isn't used anywhere, so I'm not sure what it's purpose is.

Reply:
Ha!  Good catch.